crates/transaction-pool/src/traits.rs

```diff
@@ -41,7 +41,10 @@ pub type PeerId = reth_primitives::B512;
 #[auto_impl::auto_impl(&, Arc)]
 pub trait TransactionPool: Send + Sync + Clone {
     /// The transaction type of the pool
-    type Transaction: PoolTransaction<Pooled = PooledTransactionsElementEcRecovered>;
+    type Transaction: PoolTransaction<
```

Comment:
this makes `TransactionPool` ethereum-specific, wondering if it would be better to have `trait EthTransactionPool: TransactionPool<Transaction: EthPoolTransaction>`?

Reply:
think it's ok for now, since optimism transactions can't be broadcasted, only announced over broadcast. same is done for https://github.com/paradigmxyz/reth/blob/07bf04df8a56bdc57497f9bf754947534adb16c3/crates/transaction-pool/src/ordering.rs#L33-L34

Reply:
ah you mean wrt to ongoing work of removing optimism feature from `reth_primitives::Transaction`? agreed a trait for `TransactionSignedEcRecovered` instead of `Consensus = TransactionSignedEcRecovered`, would make more sense rn.

`Pooled = PooledTransactionsElementEcRecovered` is fine for now, since op transactions aren't broadcasted just announced over broadcast, and `TransactionPool::Pooled` seems to only apply for broadcasted transactions. same is done here for example
https://github.com/paradigmxyz/reth/blob/07bf04df8a56bdc57497f9bf754947534adb16c3/crates/transaction-pool/src/ordering.rs#L22-L34

Reply:
I have the impression that this argument of `Pooled = PooledTransactionsElementEcRecovered` applies rather to the previous PR which dealt with this subject (and which we can develop if necessary in follow-ups PRs I think).

As for the `Consensus = TransactionSignedEcRecovered` which applies to quite a few places, I tried to relax as much as I could to avoid putting this constraint. Unfortunately, I noticed a cascade of constraints at this level which applies to many types and which means that we can only relax this constraint very little for the moment...